packages/next-server/lib/head.tsx

```diff
@@ -105,26 +108,23 @@ function unique() {
  *
  * @param headElement List of multiple <Head> instances
  */
-function reduceComponents(
-  headElements: Array<React.ReactElement<any>>,
-  props: WithInAmpMode
+function reduceHeadInstances(
+  headInstanceRefs: Array<React.RefObject<JSX.Element>>,
+  options: WithInAmpMode
 ) {
-  return headElements
+  return headInstanceRefs
     .reduce(
-      (list: React.ReactChild[], headElement: React.ReactElement<any>) => {
-        const headElementChildren = React.Children.toArray(
-          headElement.props.children
-        )
-        return list.concat(headElementChildren)
+      (list: Array<any>, headInstanceRef: React.RefObject<JSX.Element>) => {
+        return [...list, ...React.Children.toArray(headInstanceRef.current)]
```

Comment:
This may generate duplicate `key`s as `React.Children.toArray` / `React.Children.map` [add `key`s to elements](https://github.com/facebook/react/blob/v16.9.0/packages/react/src/ReactChildren.js#L295-L306).

This also makes the `i` in `const key = c.key || i` unreachable.